src/pages/creer-un-jeu/success.tsx

```diff
@@ -7,30 +8,73 @@ import Button from '@mui/material/Button';
 import { isGame } from 'src/activity-types/anyActivity';
 import { Base } from 'src/components/Base';
 import { ActivityContext } from 'src/contexts/activityContext';
+import { UserContext } from 'src/contexts/userContext';
+import { VillageContext } from 'src/contexts/villageContext';
 import { bgPage } from 'src/styles/variables.const';
 import PelicoSouriant from 'src/svg/pelico/pelico-souriant.svg';
+import { serializeToQueryUrl } from 'src/utils';
+import { GameType } from 'types/game.type';
 
-const PresentationSuccess: React.FC = () => {
+const PresentationSuccess = () => {
   const router = useRouter();
   const { activity } = React.useContext(ActivityContext);
+  const { village } = React.useContext(VillageContext);
+  const { axiosLoggedRequest } = React.useContext(UserContext);
+  const [firstCreation, setFirstCreation] = React.useState<boolean>(false);
 
   if (!activity || !isGame(activity)) {
     router.push('/creer-un-jeu/mimique');
   }
 
+  React.useEffect(() => {
+    if (village) {
+      axiosLoggedRequest({
+        method: 'GET',
+        url: `/games/play${serializeToQueryUrl({
+          villageId: village.id,
+          type: GameType.MIMIC,
+        })}`,
+      }).then((response) => {
+        //if user is the first one to create the mimics, then "return to homepage" is rendered
+        if (!response.error && response.data === null) {
+          setFirstCreation(response.data as boolean);
+        } else {
+          setFirstCreation(true);
+        }
+      });
+    }
+  }, [firstCreation, axiosLoggedRequest, village]);
```

Comment:
Do you need the `firstCreation` in the useEffect dependancies? If the effect updates the `firstCreation` value, there could be an infinite loop.

**Suggestion:** remove it if it's not needed.

Reply:
I removed the firstCreation  from the dependencies.